src/NServiceBus.RabbitMQ/Administration/QueueCreator.cs

```diff
@@ -50,5 +53,46 @@ void CreateQueueIfNecessary(IModel channel, string receivingAddress)
 
             routingTopology.Initialize(channel, receivingAddress);
         }
+
+        void BuildDeferralInfrastructure(IModel channel)
+        {
+            const int maxLevel = 32 - 1;
+
+            var bindingKey = "1.#";
+
+            for (int i = maxLevel; i >= 0; i--)
+            {
+                var level = i.ToString("D2");
+                var nextLevel = (i - 1).ToString("D2");
+
+                channel.ExchangeDeclare($"nsb.delay-level-{level}", ExchangeType.Topic, durableMessagesEnabled);
+
+                var arguments = new Dictionary<string, object>();
+                arguments.Add("x-queue-mode", "lazy");
+                arguments.Add("x-message-ttl", ((long)1 << i) * 1000);
+
+                var deadLetterExchangeName = i > 0 ? $"nsb.delay-level-{nextLevel}" : "nsb.delay-destination";
+                arguments.Add("x-dead-letter-exchange", deadLetterExchangeName);
+
+                channel.QueueDeclare($"nsb.delay-level-{level}", durableMessagesEnabled, false, false, arguments);
+                channel.QueueBind($"nsb.delay-level-{level}", $"nsb.delay-level-{level}", bindingKey);
+
+                bindingKey = "*." + bindingKey;
+            }
+
+            bindingKey = "0.#";
+
+            for (int i = maxLevel; i >= 1; i--)
+            {
+                var level = i.ToString("D2");
+                var nextLevel = (i - 1).ToString("D2");
+
+                channel.ExchangeBind($"nsb.delay-level-{nextLevel}", $"nsb.delay-level-{level}", bindingKey);
+
+                bindingKey = "*." + bindingKey;
```

Comment:
Is there any reason why we are not using string interpolation? As we use it above for exchange/binding/queue names